Source/Events.Store.MongoDB/EventStoreConnection.cs

```diff
@@ -40,7 +40,7 @@ public EventStoreConnection(DatabaseConnection connection, ILogger logger)
             StreamProcessorStates = connection.Database.GetCollection<StreamProcessorState>(Constants.StreamProcessorStateCollection);
             TypePartitionFilterDefinitions = connection.Database.GetCollection<TypePartitionFilterDefinition>(Constants.TypePartitionFilterDefinitionCollection);
 
-            CreateCollectionsAndIndexes();
+            CreateIndexes();
```

Comment:
When creating the indexes as we do here, Mongo implicitly creates the collection. Because we're using a mongo replica set we need to explicitly create the collections before we do any other operations on it (as far as I understand), so when the event store connection is established we want to create indexes and thus also implicitly create the collections, hence the CreateCollectionsAndIndexes name :)

Reply:
Yes we could do that